components/brave_rewards/browser/rewards_service.h

```diff
@@ -46,6 +46,7 @@ bool IsMediaLink(const GURL& url,
 
 class RewardsNotificationService;
 class RewardsServiceObserver;
+class RewardsServicePrivateObserver;
```

Comment:
why do we need another observer and what is `RewardsServicePrivateObserver` vs `RewardsServiceObserver`? We can't just keep adding stuff without comments, especially when it's not obvious at all what the difference is without digging through a lot of code

Reply:
it will be investigated in:
https://github.com/brave/brave-browser/issues/6337

Reply:
we should remove this code if we are not sure about it (like brave/brave-browser#6337 says) and add it back/modify when we know the whole picture. We shouldn't be adding code just because

Reply:
we should create issue to add it back/modify it, not to investigate why we have it

Reply:
@NejcZdovc it is not just because, we use it to handle events that you handle here in desktop browser https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/extension_rewards_service_observer.h. If we remove it the callbacks update functionality will be broken, the task is to investigate that and improve if we don't need it. Desktop panel is done via extension, it is a different way on Android and the current approach doesn't work.

Reply:
that's why @bridiver asked why you can't just use `RewardsServiceObserver` as you don't have extensions so you don't need `ExtensionRewardsServiceObserver` like we do on desktop

Reply:
@NejcZdovc I don't get what are you trying to achieve. It was discussed we create issues on NON CRITICAL things and fix CRITICAL things. It is not CRITICAL and wasn't asked by you.

Reply:
Why do you make noise to that PR by comments you post?

Reply:
I don't see @bridiver said that he is ok to be merged like this. Why I am commenting on it, because would like to resolve it asap and merge it

Reply:
Critical issues were marked with a word `critical` and discussed in the internal chat https://bravesoftware.slack.com/archives/CJVNBPK9P/p1569618124085000?thread_ts=1569602718.050800&cid=CJVNBPK9P